ddtrace/span.py

```diff
@@ -457,6 +458,13 @@ def _remove_exc_info(self):
     def pprint(self):
         # type: () -> str
         """ Return a human readable version of the span. """
+
+        def _dict_section(name, data):
+            # type: (str, Union[Dict[str, Any], Dict[Union[bytes, str], Any]]) -> List[Tuple[str, str]]
```

Comment:
```suggestion
            # type: (str, Dict[Union[bytes, str], Union[str, bytes, float, int]]) -> List[Tuple[str, str]]
```

? These should be the only options available.

The key will always be bytes or str (just str?)

And the value can only be a bytes, str, float, or int.

Reply:
`mypy` doesn't like `Dict[Union[bytes, str]` for the second argument, given that `self.metrics` and `self.meta` have different types and this helper is invoked on both. That's why I left a comment for @majorgreys, who has recently looked into these types, to see whether it would make sense to use the same type for both. But now I see that they are actually different 🤔

Reply:
we just need to update this type hinting with the new types and this PR should be good to go.